### PR TITLE
Alor e/student/candidate features

### PR DESCRIPTION
**Pull request by Alor-e**
I added the Candidate class which inherits from the User class and also the Application Class which relies on Candidates, Jobs, Resume and Status.

Resume and Status would come next.

### CODE REVIEW

**Comment by Alor-e** [comment]
This is related to Issue #19 #21

File: be-career-platform/server/model.py [hunk 1]
```diff
@@ -121,3 +121,123 @@ def playloadToInsert(self):
...
+        }
+
+    def save_to_mongo(self):
+        mongo.db.applications.insert_one(self.json())
+
+
+class Status():
+    # define some constants for the status values
+    PENDING = "pending"
+    ACCEPTED = "accepted"
```
Comment (Elsavid) [inline be-career-platform/server/model.py:237]: Employers should be able to offer an interview, so I think "accepted" status need to be changed.

Comment (Alor-e) [reply be-career-platform/server/model.py:237]: Okay I will update that

Comment (Alor-e) [reply be-career-platform/server/model.py:237]: maybe we can make it 4 status values not 3

Comment (Alor-e) [reply be-career-platform/server/model.py:237]: `PENDING = "pending"
    INTERVIEW = "interview"
    ACCEPTED = "accepted"
    REJECTED = "rejected"`

**Review by paullmq8 — APPROVED** [review]
good

**Pull request merged**